src/main/java/org/jenkinsci/plugins/workflow/steps/WaitForConditionStep.java

```diff
@@ -36,13 +36,28 @@
 import javax.annotation.Nonnull;
 import jenkins.util.Timer;
 import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
 
 public final class WaitForConditionStep extends Step {
 
+    static final long MIN_RECURRENCE_PERIOD = 250; // ¼s
+    static final long MAX_RECURRENCE_PERIOD = 15000; // ¼min
+
+    private long initialRecurrencePeriod = MIN_RECURRENCE_PERIOD;
+
     @DataBoundConstructor public WaitForConditionStep() {}
 
+    @DataBoundSetter
+    public void setInitialRecurrencePeriod(long initialRecurrencePeriod) {
+        this.initialRecurrencePeriod = Math.min(initialRecurrencePeriod, MAX_RECURRENCE_PERIOD);
```

Comment:
I think you also need to make sure this is at least `MIN_RECURRENCE_PERIOD`:
```suggestion
        this.initialRecurrencePeriod = Math.max(MIN_RECURRENCE_PERIOD, Math.min(initialRecurrencePeriod, MAX_RECURRENCE_PERIOD));
```